src/fsharp/ast.fs

```diff
@@ -1944,19 +1944,21 @@ let mkSynApp3 f x1 x2 x3 m = mkSynApp1 (mkSynApp2 f x1 x2 m) x3 m
 let mkSynApp4 f x1 x2 x3 x4 m = mkSynApp1 (mkSynApp3 f x1 x2 x3 m) x4 m
 let mkSynApp5 f x1 x2 x3 x4 x5 m = mkSynApp1 (mkSynApp4 f x1 x2 x3 x4 m) x5 m
 let mkSynDotParenSet  m a b c = mkSynTrifix m parenSet a b c
-let mkSynDotBrackGet  m mDot a b   = SynExpr.DotIndexedGet (a, [SynIndexerArg.One b], mDot, m)
+let mkSynDotBrackGet  m mDot a b fromEnd   = SynExpr.DotIndexedGet (a, [SynIndexerArg.One (b, fromEnd, m)], mDot, m)
 let mkSynQMarkSet m a b c = mkSynTrifix m qmarkSet a b c
 let mkSynDotBrackSliceGet  m mDot arr sliceArg = SynExpr.DotIndexedGet (arr, [sliceArg], mDot, m)
 
 let mkSynDotBrackSeqSliceGet  m mDot arr (argsList: list<SynIndexerArg>) =
-    let notSliced=[ for arg in argsList do
-                       match arg with
-                       | SynIndexerArg.One x -> yield x
-                       | _ -> () ]
-    if notSliced.Length = argsList.Length then
-        SynExpr.DotIndexedGet (arr, [SynIndexerArg.One (SynExpr.Tuple (false, notSliced, [], unionRanges (List.head notSliced).Range (List.last notSliced).Range))], mDot, m)
-    else
-        SynExpr.DotIndexedGet (arr, argsList, mDot, m)
+    // We want to be able to turn .[1,2,3] into a Tuple(1,2,3) but keep 
```

Comment:
Commented code is normally deleted. Git keeps track of history.

Reply:
oops